classes/renderables/course_card.php

```diff
@@ -249,10 +249,19 @@ private function apply_contact_avatars() {
         }
 
         $avatars = array_merge($avatars, $blankavatars);
-        // Cached value of avatar array for the course.
-        $avatarcache->set($this->contextid, $avatars);
-        // Context ID + User ID index for handling unenrolments and deletions.
-        $indexcache->set('idx', $userctxidx);
+
+        if (!empty($avatars)) {
```

Comment:
Only cache when avatars aren't empty. Caching an empty array caused an error for me but I can't remember how to replicate the bug.